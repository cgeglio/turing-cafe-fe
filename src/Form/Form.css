input {
  margin: 10px;
  height: 20px;
  text-indent: 5px;
  width: 160px;
}

.submit-btn {
  border: 1px solid white;
  background-color: #028080;
  color: white;
  cursor: pointer;
  height: 20px;
  padding: 1px;
  width: 100px;
}

.submit-btn:hover {
  border: 2px solid #028080;
  background-color: white;
  color: #028080;
  padding: 0;
}
