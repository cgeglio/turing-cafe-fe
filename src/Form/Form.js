import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor() {
    super();
    this.state = {name: '', date: '', time: '', number: ''}
  }

  handleChange = event => {
    this.setState({[event.target.name]: event.target.value})
  }

  addReservation = event => {
    event.preventDefault();
    this.props.addReservation({name: this.state.name, date: this.state.date, time: this.state.time, number: this.state.number})
    this.setState({name: '', date: '', time: '', number: ''});
  }

  render() {
    return (
      <form>
        <input
          type="text"
          placeholder="Name..."
          name="name"
          value={this.state.name}
          onChange={this.handleChange}
        />
        <input
          type="text"
          placeholder="01/01"
          name="date"
          value={this.state.date}
          onChange={this.handleChange}
        />
        <input
          type="text"
          placeholder="12:00"
          name="time"
          value={this.state.time}
          onChange={this.handleChange}
        />
        <input
          type="text"
          placeholder="Number of people..."
          name="number"
          value={this.state.number}
          onChange={this.handleChange}
        />
        <button className="submit-btn" onClick={this.addReservation}>Submit</button>
      </form>
    )
  }
}

export default Form;
