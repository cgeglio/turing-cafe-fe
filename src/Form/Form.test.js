import React from 'react';
import ReactDOM from 'react-dom';
import Form from './Form';
import { shallow } from 'enzyme';

describe('Form', () => {
  let wrapper;
  let mockAddReservation = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Form
      addReservation={mockAddReservation}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should set state when handleChange is called', () => {
    const mockEvent = { target: { name: 'name', value: 'Eric'} };
    const expected = 'Eric';

    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('name')).toEqual(expected);
  });

  it('should set reset state when addReservation is called', () => {
    const mockEvent = { preventDefault: jest.fn() };
    const mockState = {name: 'Eric', date: '01/10', time: '10:00', number: 5};
    const expected = {name: '', date: '', time: '', number: ''};

    wrapper.instance().setState(mockState)
    wrapper.instance().addReservation(mockEvent);
    expect(wrapper.state()).toEqual(expected);
  });

  it('should call addReservation when the submit button is clicked', () => {
    wrapper.instance().addReservation = jest.fn();
    wrapper.instance().forceUpdate();
    const mockEvent = { preventDefault: jest.fn() };

    wrapper.find('button').simulate('click', mockEvent);
    expect(wrapper.instance().addReservation).toHaveBeenCalledWith(mockEvent);
  });



});
