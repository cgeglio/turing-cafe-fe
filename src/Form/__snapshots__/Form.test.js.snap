// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<form>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name..."
    type="text"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="01/01"
    type="text"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="12:00"
    type="text"
    value=""
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of people..."
    type="text"
    value=""
  />
  <button
    className="submit-btn"
    onClick={[Function]}
  >
    Submit
  </button>
</form>
`;
